lib/game/game_setup.ex

```diff
@@ -20,20 +22,30 @@ defmodule Game.GameSetup do
     Players.set_players(player_one, player_two)
   end
 
-  @spec select_player_type(String.t()) :: player
-  defp select_player_type(player_selection) do
+  @spec select_player_type(String.t(), list()) :: player
+  defp select_player_type(player_selection, player_types) do
     {selection_number, _selection_decimal} = Integer.parse(player_selection)
-    %{type: player_type} = Enum.at(@player_types, selection_number - 1)
+    %{type: player_type} = Enum.at(player_types, selection_number - 1)
     player_type
   end
 
+  defp select_difficulty(HumanPlayer, _interface), do: HumanPlayer
+  defp select_difficulty(computer_types, interface) do
+    computer_types
+    |> interface.formatter.format_computer_difficulty_setup()
+    |> interface.communicator.read_input()
+    |> String.trim()
```

Comment:
I think its always called with `read_input()`. Funnily enough, I just tried removing all of the `String.trim()` and everything stayed green and the `mix start` ran fine, should I just remove it all together or should I package it and  `read_input()` together?

Reply:
If everything works without it do you need it? I'm a little surprised by that since I thought you tried removing it before and tests did fail.

If you don't need it I would remove it, if you do need it I would move it into the interface in some way so you don't need to remember to call it everywhere `read_input` is used.